rerun_py/rerun_sdk/rerun/send_columns.py

```diff
@@ -182,15 +182,21 @@ def send_columns(
             indicators.append(c)
             continue
         component_name = c.component_name()
-        component_column = c.as_arrow_array()  # type: ignore[union-attr]
+
+        if isinstance(c, ComponentColumn):
+            component_column = c
+        else:
+            component_column = c.partition([1] * len(c))  # type: ignore[union-attr, attr-defined, arg-type]
```

Comment:
As far as I understand, the only way to have access to `partition()` is to be a `ComponentBatch{Like}`, which is what's assumed here -- but it probably would be nicer both for the reader and the caller if it was explicitly stated (also would probably get rid of these type hints).

```suggestion
        elif istance(c, ComponentBatch):
            component_column = c.partition([1] * len(c))
        else:
            raise ValueError("blabla")